calour/ms1_experiment.py

```diff
@@ -196,3 +161,129 @@ def get_bad_features(self, mz_tolerance=0.001, rt_tolerance=2, corr_thresh=0.8,
                 keep_features.extend(ckeep)
                 features.drop(index=ckeep, inplace=True)
         return self.filter_ids(keep_features, negate=negate, inplace=inplace)
+
+    def merge_similar_features(self, mz_tolerance=0.001, rt_tolerance=0.5):
+        '''Merge metabolites with similar mz/rt to a single metabolite
+
+        metabolites are initially sorted by frequency and a greefy clustering algorithm (starting from the highest freq.) is used to join together
+        metabolites that are close in m/z and r/t.
+
+        Parameters
+        ----------
+        mz_tolerance: float, optional
+            metabolites with abs(metabolite_mz - mz) <= mz_tolerance are joined
+        rt_tolerance: float, optional
+            metabolites with abs(metabolite_rt - rt) <= rt_tolerance are joined
+
+        Returns
+        -------
+        calour.MS1Experiment with  close metabolites joined to a single metabolite.
+        The m/z and rt of the new metabolite are the m/z and rt of the highest freq. metabolite.
+        new feature_metadata fields: _calour_merge_number, _calour_merge_ids are added listing the number and ids of the metabolites joined for each new metabolite
+        '''
+        exp = self.sort_abundance(reverse=False)
+        features = exp.feature_metadata
+        features['_metabolite_group'] = np.zeros(len(features)) - 1
```

Comment:
instead of of messing and clutter feature metadata data frame, could you modify `aggregate_by_metadata` to accept a list-like (pd.series, tuple, etc.) besides a column name for group-then-apply manipulation? it is a very light change and avoids adding intermediary column to metadata data frame.

Reply:
do you mean to make the field var in agg_be metadata can be a list/pd.series etc?
I think this will be confusing for the API of the function (field can be a tuple...)
adding it as a different param instead of field will also be confusing for the api...
The new column is deleted from the final result, so the user does not feel anything.
i think making the API clean is more important than making the code clean...